packages/engine-server/src/config.ts

```diff
@@ -247,7 +247,10 @@ export class DConfig {
   static readConfigSync(wsRoot: string) {
     const configPath = DConfig.configPath(wsRoot);
     // TODO: validate
```

Comment:
if we are using this as the source of truth, it should have logic of `readConfigAndApplyLocalOverrideSync`

Reply:
also, as we discussed offline, there are really multiple configs at play here.

the workspace config, the local config override (if it exists), and the combined config. would recommend returning an object here 

{mergedConfig, wsConfig, localConfig} so that operations that need to update the config can update the `wsConfig`

Reply:
Can we do this in a separate pr? I wanted to first move all usages of engine.config to Dconfig first before introducing any new business logic. Otherwise it might be too many moving pieces at once